__tests__/app.test.js

```diff
@@ -61,66 +62,119 @@ describe("GET /api/articles/:article_id", () => {
+          expect(article).not.toHaveProperty("body");
+        });
+      });
+  });
+});
+describe("GET /api/articles/:article_id/comments", () => {
+  test("GET:200 sends an array of comments for a single article to the client", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then((response) => {
+        const comments = response.body.comments;
+        expect(comments).toHaveLength(2);
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number),
+          });
         });
       });
-    });
-  })
-  test('GET:200 sends an array of articles to the client sorted by date in descending order', () => {
+  });
+  test("GET:200 sends an array of comments for a single article to the client with the most recent comments first", () => {
     return request(app)
-    .get("/api/articles")
-    .expect(200)
-    .then((response) => {
-      const articles = response.body.articles;
-      console.log(articles)
-      expect(articles).toHaveLength(13);
-      expect(articles).toBeSortedBy("created_at", { descending: true });
-    });
-  })
-  test('GET:200 sends an array of articles to the client which do not include the articles body column', () => {
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        const comments = response.body.comments;
+        expect(comments).toHaveLength(11);
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("GET:404 reponds with an appropriate error message when given a valid but non-existent id", () => {
+    return request(app)
+      .get("/api/articles/34324/comments")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("entry does not exist");
+      });
+  });
+  test("GET:400 responds with an appropriate error message when given an invalid id", () => {
     return request(app)
-    .get("/api/articles")
-    .expect(200)
-    .then((response) => {
-      const articles = response.body.articles;
-      console.log(articles)
-      articles.forEach((article) => {
-        expect(article).not.toHaveProperty('body');
-      })     
-    });
-  })
-})
+      .get("/api/articles/not-an-id/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad Request");
+      });
+  });
+  test("GET:200 responds with an appropriate  message when given an article id for an article which does not have any comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.msg).toBe("no comments for this article");
```

Comment:
Nearly there, but in this instance we would need to test that an empty array is returned also